### PR TITLE
Project cleanup

### PR DESCRIPTION
**Pull request by chrisb87**
This is some cleanup and organization to make developing on the project easier.
* Ruby files moved to /app, cantaloupe and mysql connection jar moved to /lib, both directories are copied into the Docker image.
* Additional ruby files can now be added to /app, required in delegates.rb, used in the app without additional changes to build scripts.
* Rather than rebuilding the Docker image with every change, updates to the ruby scripts are hot-reloaded when developing with docker compose
* Removed several unused files

This requires a change to the bamboo scripts or the deployment will break!

For QA, the third of the four bamboo tasks can be shortened to:
```bash
echo $bamboo_iplist > list

for i in `cat list`
do

ssh sysadmin@$i "sudo systemctl stop cantaloupe"
sleep 30

ssh sysadmin@$i "cp /var/www/images.nypl.org/app/* /var/www/images.nypl.org/cantaloupe"
ssh sysadmin@$i "cp /var/www/images.nypl.org/cantaloupe.properties /var/www/images.nypl.org/cantaloupe.properties.old"
ssh sysadmin@$i "cp /var/www/images.nypl.org/app/cantaloupe.properties.dev /var/www/images.nypl.org/cantaloupe.properties"

ssh sysadmin@$i "sudo systemctl start cantaloupe"

sleep 30

done

```

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -8,20 +8,21 @@ services:
...
+  # db:
+  #   image: mysql
+  #   command: --default-authentication-plugin=mysql_native_password
+  #   environment:
+  #     MYSQL_ROOT_PASSWORD: password
+  #   volumes:
+  #     - ./db/data:/var/lib/mysql
+  #     - ./db/setup:/docker-entrypoint-initdb.d
+  #   ports:
+  #     - 3306:3306
```
Comment (croyfish) [inline docker-compose.yml:28]: Did you mean to keep the shim commented out?

Comment (chrisb87) [reply docker-compose.yml:28]: I should probably revert that

Comment (chrisb87) [reply docker-compose.yml]: done!

**Comment by croyfish** [comment]
Both the image server and the shim are working fine, and I really like the ability to hot reload rubycode. How difficult is it going to be to get those pipeline updates in?

**Comment by chrisb87** [comment]
> Both the image server and the shim are working fine, and I really like the ability to hot reload rubycode. How difficult is it going to be to get those pipeline updates in?

Hard to know how difficult it would be but I don't imagine it being an issue. Stevie has been helpful updating these in the past. We may be able to ourselves. It should be relatively harmless to try things out in QA first to make sure it's smooth.

**Review by croyfish — APPROVED** [review]

**Comment by emu47** [comment]
Net -13k lines of code is nice. :)

**Comment by chrisb87** [comment]
Closing this but using it as a reference for the work in https://newyorkpubliclibrary.atlassian.net/browse/DR-3118

**Pull request closed without merging**